init/bootstrap.go

```diff
@@ -43,10 +46,18 @@ outer:
 			continue
 		}
 
-		for _, b := range buffer {
-			if b != 0 {
-				log.Infof("%s not empty", dev)
-				continue outer
+		boot2docker := false
+
+		if strings.HasPrefix(string(buffer), boot2dockerMagic) {
```

Comment:
so you're saying, If i change the mutable buffer object, the string will change as well, right?